prow/cmd/pipeline/controller.go

```diff
@@ -613,24 +617,39 @@ func reconcile(c reconciler, key string) error {
 		return fmt.Errorf("no pipelinerun found or created for %q, wantPipelineRun was %v", key, wantPipelineRun)
 	}
 	wantState, wantMsg := prowJobStatus(runs)
-	return updateProwJobState(c, pj, wantState, wantMsg)
+	return updateProwJobState(c, pj, wantState, wantMsg, ctx, namespace, runs)
 }
 
-func updateProwJobState(c reconciler, pj *prowjobv1.ProwJob, state prowjobv1.ProwJobState, msg string) error {
+func updateProwJobState(c reconciler, pj *prowjobv1.ProwJob, state prowjobv1.ProwJobState, msg, context, namespace string, runs []*pipelinev1alpha1.PipelineRun) error {
 	haveState := pj.Status.State
 	haveMsg := pj.Status.Description
 	if haveState != state || haveMsg != msg {
 		npj := pj.DeepCopy()
-		if npj.Status.StartTime.IsZero() {
-			npj.Status.StartTime = c.now()
-		}
-		if npj.Status.CompletionTime.IsZero() && finalState(state) {
-			now := c.now()
-			npj.Status.CompletionTime = &now
+		if state == prowjobv1.FailureState && strings.Contains(msg, pipelineRunShouldRetryMessage) {
+			// If the pipeline failed due to the known race condition, wipe the status and retry.
+			npj.Status = prowjobv1.ProwJobStatus{
```

Comment:
I would try to set up the `DeletionTImestamp` and let the normal loop to cleanup the pipelineruns and afterwards reset the status to retrigger. 

This routine is only meant to update the status and not really to do cleanups.

Reply:
Ok, lemme figure that out.